qiskit_experiments/randomized_benchmarking/rb_experiment.py

```diff
@@ -0,0 +1,128 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Standard RB Experiment class.
+"""
+from typing import Union, Iterable, Optional
+
+import numpy as np
+from numpy.random import Generator, default_rng
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info import Clifford, random_clifford
+
+from qiskit_experiments.base_experiment import BaseExperiment
+from .rb_analysis import RBAnalysis
+
+
+class RBExperiment(BaseExperiment):
+    """RB Experiment class"""
+
+    # Analysis class for experiment
+    __analysis_class__ = RBAnalysis
+
+    def __init__(
+        self,
+        qubits: Union[int, Iterable[int]],
+        lengths: Iterable[int],
+        num_samples: int = 1,
+        seed: Optional[Union[int, Generator]] = None,
+    ):
+        """Standard randomized benchmarking experiment
+
+        Args:
+            qubits: the number of qubits or list of
+                    physical qubits for the experiment.
+            lengths: A list of RB sequences lengths.
+            num_samples: number of samples to generate for each
+                         sequence length
+            seed: Seed or generator object for random number
+                  generation. If None default_rng will be used.
+        """
+        if not isinstance(seed, Generator):
+            self._rng = default_rng(seed=seed)
+        else:
+            self._rng = seed
+        self._lengths = list(lengths)
+        self._num_samples = num_samples
+
+        super().__init__(qubits)
+
+    # pylint: disable = arguments-differ
+    def circuits(self, backend=None):
+        """Return a list of RB circuits.
+
+        Args:
+            backend (Backend): Optional, a backend object.
+
+        Returns:
+            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s.
+        """
+        circuits = []
+        for length in self._lengths:
+            for _ in range(self._num_samples):
+                circ = self._sample_circuit(length, seed=self._rng)
```

Comment:
Currently in RB, we do not generate independent circuits for each length; we start with a circuit of length[0], create a copy with inverse and measurements, and then continue to add cliffords to the same circuit until we reach length[1], add inverse and measurement, etc.

We might want to keep this behavior to speed up the runtime and maintain backwards-compatibility, although I'm not sure how critical both of those are.

Reply:
I guess that explains part of why this version was so much slower. I think it would be good to allow an option for both, but probably the default is how it was done previously to be faster.